Source/UI/BillingForm/View/BillingFormTextFieldView.swift

```diff
@@ -5,6 +5,37 @@ protocol PhoneNumberTextFieldDelegate: AnyObject {
     func phoneNumberIsUpdated(number: String, tag: Int)
 }
 
+class BorderView: UIView {
+
+    private var style: ElementBorderStyle?
+
+    /// shaper layer that draw edges and corners
+    var borderLayer = CAShapeLayer()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        layer.addSublayer(borderLayer)
+    }
+
+    // Keep border layer size in sync with owning view
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        guard let style = style else { return }
+        borderLayer.frame = bounds
+        borderLayer.updateEdgesAndCorners(with: style)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    /// Update edges and corners style
+    /// - Parameter style: protocol with some parameters for styling border and corners
+    func update(with style: ElementBorderStyle) {
```

Comment:
We could add another method
```swift
func updateBorderColor(to newBorderColor: UIColor) {
  // update
}
```

which would allow us to keep the layer private.

Reply:
Also, this method behaves more like a `setup` that an update. As in, it will only ever accept first style provided.

We should either document this limitation, or try and support the behaviour as described:
```swift
if self.style != style {
    self. borderLayer = nil
    self.style = style
    self.layoutSubviews()
}
```